client/src/App.js

```diff
@@ -13,16 +14,28 @@ export default class App extends Component {
   }
 
   addToSavedList = movie => {
+    console.log(this.state.savedList);
     const savedList = this.state.savedList;
-    savedList.push(movie);
-    this.setState({ savedList });
+    if (this.state.savedList.includes(movie)) {
+      alert(`${movie.title} is already saved`);
+    } else {
+      savedList.push(movie);
+      this.setState({ savedList });
+    }
   };
 
   render() {
     return (
       <div>
         <SavedList list={this.state.savedList} />
-        <div>Replace this Div with your Routes</div>
+        <Route exact path="/" component={MovieList} />
+        <Route
+          exact
```

Comment:
You can drop the exact attribute here, but if you had another route that went to just `/movies`  it'd be a good idea to make that route `exact` to differentiate it from the individual `movies/:id` routes.  The order is also important in terms of your routes when using exact so it's good to be aware of that in future when you'll be building more complex navigational components 👍 

[<Route exact docs section>](https://reacttraining.com/react-router/core/api/Route/exact-bool)